index.bs

```diff
@@ -3069,14 +3071,25 @@ The [=remote end steps=] with |command parameters| are:
 
 1. Let |viewport parameter| be the |command parameters|["<code>viewport</code>"] field.
 
+1. Let |device pixel ratio parameter| be the |command parameters|["<code>devicePixelRatio</code>"] field.
+
+1. If |device pixel ratio parameter| is not null and the implementation is not able to change the size of the [=CSS Pixel=] in |context|'s [=layout viewport=] for any reason, return [=error=] with [=error code=] [=unsupported operation=].
```

Comment:
I think this should say "to the specified values" or something to indicate that it's not only if you're unable to change it in general, but specifically if you're unable to change it to the parameters given. It's a bit unfortunate to say that without saying what the values represent. An alternative, but less concise, phrasing would be:
```
1. Let |viewport| be |command parameters| ["<code>viewport</code>"]

1. Let |device pixel ratio| be |command parameters|["<code>devicePixelRatio"</code>]

1. If |viewport| is not null, and the implementation is unable to set the width and height dimensions
   of the [=layout viewport=] to |viewport|["width"] and |viewport|["height"], respectively, or if 
   |device pixel ratio| is not null and the implementation is unable to set the size of the CSS pixel 
   of the [=layout viewport=] to |device pixel ratio| device pixels, or if the implementation is unable
   to adjust the [=layout viewport=] parameters for any other reason, return [=error=] 
   with [=error code=] [=unsupported operation=]
```

Reply:
Thanks, I have updated to the alternative phrasing you propose.